crates/iceberg/src/spec/schema.rs

```diff
@@ -376,6 +375,24 @@ impl Schema {
     pub fn accessor_by_field_id(&self, field_id: i32) -> Option<Arc<StructAccessor>> {
         self.field_id_to_accessor.get(&field_id).cloned()
     }
+
+    /// Check if this schema is identical to another schema semantically - excluding schema id.
+    pub(crate) fn is_same_schema(&self, other: &SchemaRef) -> bool {
```

Comment:
I thought about this as well, but opted for a method with documentation.
This method excludes the `schema_id` which I can describe here in the docstring. Using `Eq` I would expect the `schema_id` to be equal too - especially because its used in tests.